controllers/applicationvolumereplication_controller.go

```diff
@@ -195,658 +276,895 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
+	if err != nil {
+		return nil, nil, err
+	}
 
-	return ctrl.Result{Requeue: requeue}, nil
+	return avrPlRule, userPlRule, nil
 }
 
-type AVRInstance struct {
-	reconciler       *ApplicationVolumeReplicationReconciler
-	ctx              context.Context
-	log              logr.Logger
-	instance         *rmn.ApplicationVolumeReplication
-	subscriptionList *subv1.SubscriptionList
-}
-
-// For each subscription
-//		Check if it is paused for failover
-//			- restore PVs to the failed over cluster
-// 			- unpause
-//          - go to next subscription
-//		otherwise, select placement decisions
-//			- extract home cluster from placementrule.status.decisions
-//			- extract peer cluster from the clusters forming the dr pair
-//				example: ManagedCluster Set {A, B, C, D}
-//						 Pl.GenericPlacementField results in DR_Set = {A, B}
-//						 plRule{Status.Decision=A}
-//						 homeCluster = A
-//						 peerCluster = (DR_Set - A) = B
-//		create or update ManifestWork
-// returns placement decisions which can be the decisions for only a subset of subscriptions
-//
-func (a *AVRInstance) processSubscriptions() bool {
-	a.log.Info("Process subscriptions", "total", len(a.subscriptionList.Items))
+func (r *ApplicationVolumeReplicationReconciler) getUserPlacementRule(ctx context.Context,
+	avr *rmn.ApplicationVolumeReplication) (*plrv1.PlacementRule, error) {
+	r.Log.Info("Getting User PlacementRule", "placement", avr.Spec.Placement)
 
-	requeue := false
+	if avr.Spec.Placement == nil || avr.Spec.Placement.PlacementRef == nil {
+		return nil, fmt.Errorf("invalid user placementRule for AVR %s", avr.Name)
+	}
 
-	for idx, subscription := range a.subscriptionList.Items {
-		// On the hub ignore any managed cluster subscriptions, as the hub maybe a managed cluster itself.
-		// SubscriptionSubscribed means this subscription is child sitting in managed cluster
-		// Placement.Local is true for a local subscription, and can be used in the absence of Status
-		if subscription.Status.Phase == subv1.SubscriptionSubscribed ||
-			(subscription.Spec.Placement != nil && subscription.Spec.Placement.Local != nil &&
-				*subscription.Spec.Placement.Local) {
-			a.log.Info("Skipping local subscription", "name", subscription.Name)
+	plRef := avr.Spec.Placement.PlacementRef
 
-			continue
-		}
+	if plRef.Namespace == "" {
+		plRef.Namespace = avr.Namespace
+	}
 
-		placementDecision, needRequeue := a.processSubscription(&a.subscriptionList.Items[idx])
+	userPlacementRule := &plrv1.PlacementRule{}
 
-		if needRequeue {
-			a.log.Info("Requeue for subscription", "name", subscription.Name)
+	err := r.Client.Get(ctx,
+		types.NamespacedName{Name: plRef.Name, Namespace: plRef.Namespace}, userPlacementRule)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get placementrule error: %w", err)
+	}
 
-			requeue = true
+	scName := userPlacementRule.Spec.SchedulerName
+	if scName != "" && scName != RamenScheduler {
```

Comment:
If scName is "", then we would still process it as Ramen scheduled, based on the condition `scName != ""` being false. So we just need a check against scName as RamenScheduler?